### ISSUE TITLE
Bug: wrapping a templated static method when defined outside of class

### ISSUE THREAD

**Issue by schnoerr14** [swig/swig#1754]
Hello, to illustrate the issue I wrote two classes A and B, both with a static method f().

The problem arises with A::f( Tnum i ), as opposed to B::f( Tnum i ), since the function body is defined outside of class A. 

The error message:
```
Test_StaticTemplateMethod_check_wrap.cxx: In function ‘PyObject* _wrap_A_f_i(PyObject*, PyObject*)’:
Test_StaticTemplateMethod_check_wrap.cxx:2960:17: error: ‘A_f’ was not declared in this scope
   result = (int)A_f< int >(arg1);
```
If here 'result = (int)A::f<int>(arg1)' was generated all would work fine.

After inspection of the resulting wrapper-code I assume, that swig includes the source of the static member function instead of purely relying on its API, and this fails when the method definition is separated of its declaration inside the class. This is a problem at least in cases where the code is separated in a declaration header .h and an implementation file .cc.

By comparison with B::f<int> the wrapper code is:
SWIGINTERN int B_f_Sl_int_Sg_(int i){ std::cout << "B::f<int>( i= " << i << " );" << std::endl; }
result = (int)B_f_Sl_int_Sg_(arg1);
This SWIGINTERN... line is missed for class A.

I'm using swig 4.0.1 with g++-7.4.0

Here the code of the condensed example:
-----------------------------------
```
#include <iostream>
using namespace std;

class A {
public:
    A() {}
    template <typename Tnum>
    static inline Tnum f( Tnum i );
};
template <typename Tnum>
Tnum A::f( Tnum i ) { std::cout << "A::f<int>( i= " << i << " );" << std::endl; }

class B {
public:
    B() {}
    template <typename Tnum>
    static inline Tnum f( Tnum i ) { std::cout << "B::f<int>( i= " << i << " );" << std::endl; }
};
```
The Interface:
--------------------------------
```
%module Test_StaticTemplateMethod_check

%header %{
#include "Test_StaticTemplateMethod.h"
%}
%include "Test_StaticTemplateMethod.h"

%extend A {
    %template(f_i) A::f<int>;
 };
%extend B {
    %template(f_i) B::f<int>;
 };
```

**Comment by schnoerr14** [comment]
Hello again,
as I see the code I pasted was garbled (template <...>)
I attach my report as a .txt-file again.
[20200307_Test_StaticTemplateMethod_check.txt](https://github.com/swig/swig/files/4301767/20200307_Test_StaticTemplateMethod_check.txt)

**Comment by ojwb** [comment]
Still present in current git master.

Here's an all-in-one reproducer without GCC -Wreturn-type warnings:

```
%module test

%inline %{
#include <iostream>

class A {
public:
    A() {}
    template <typename Tnum>
    static inline Tnum f( Tnum i );
};

template <typename Tnum>
Tnum A::f( Tnum i ) {
    std::cout << "A::f<int>( i= " << i << " );" << std::endl;
    return i;
}

class B {
public:
    B() {}
    template <typename Tnum>
    static inline Tnum f( Tnum i ) {
        std::cout << "B::f<int>( i= " << i << " );" << std::endl;
        return i;
    }
};
%}

%extend A {
    %template(f_i) A::f<int>;
 };
%extend B {
    %template(f_i) B::f<int>;
 };
```

**Comment by ojwb** [comment]
Hiding the definition of `A::f` from SWIG with `#ifndef SWIG` doesn't seem to help, so it appears the problem is specifically in how SWIG handles the declaration of `A::f` without a body compared to `B::f` with a body.

**Comment by ojwb** [comment]
Fixed by feaabb0d2a90635642768aa1da1439f5a9a9b874.
